usecases/evaluate_scenario/evaluate_sanction_check.go

```
@@ -70,6 +70,37 @@ func (e ScenarioEvaluator) evaluateSanctionCheck(
 		return
 	}
 
+	if iteration.SanctionCheckConfig.WhitelistField != nil {
+		whitelistFieldResult, err := e.evaluateAstExpression.EvaluateAstExpression(
+			ctx,
+			nil,
+			*iteration.SanctionCheckConfig.WhitelistField,
+			params.Scenario.OrganizationId,
+			dataAccessor.ClientObject,
+			params.DataModel,
+		)
+		if err != nil {
+			sanctionCheckErr = errors.Wrap(err, "could not extract object field for whitelist check")
+			return
+		}
+
+		whitelistField, err := whitelistFieldResult.GetStringReturnValue()
```

Here, IMO we should adapt logic if a null value is read (check on ErrNullFieldRead returned, or just rewrite the downcast logic here) as well as if an empty string is read.
(well: empty string perhaps takes care of itself in FilterOut, but at least avoid errors incase of null)

---

I guess we only want to throw an error for unexpected `errs` and continue processing (but skipping registering the whitelist) on `ErrNullFieldRead` and `whitelistField == ""`?

---

yes agreed with this